src/gentropy/datasource/gwas_catalog/associations.py

```diff
@@ -570,175 +569,107 @@ def _are_alleles_palindromic(
-            p_value (Column): GWAS Catalog p-value column
-            direction (str): This is the direction of the confidence interval. It can be either "upper" or "lower".
-
-        Returns:
-            Column: The upper and lower bounds of the confidence interval for the beta coefficient.
-        """
-        zscore_95 = f.lit(1.96)
-        beta = GWASCatalogCuratedAssociationsParser._harmonise_beta(
-            risk_allele,
-            reference_allele,
-            alternate_allele,
-            effect_size,
-            confidence_interval,
-        )
-        zscore = pvalue_to_zscore(p_value)
-        return (
-            f.when(f.lit(direction) == "upper", beta + f.abs(zscore_95 * beta) / zscore)
-            .when(f.lit(direction) == "lower", beta - f.abs(zscore_95 * beta) / zscore)
-            .otherwise(None)
-        )
-
     @staticmethod
     def _harmonise_odds_ratio(
-        risk_allele: Column,
-        reference_allele: Column,
-        alternate_allele: Column,
         effect_size: Column,
-        confidence_interval: Column,
+        flipping_needed: Column,
     ) -> Column:
-        """Harmonizing odds ratio.
+        """Odds ratio is either propagated as is, or flipped if indicated, meaning returning a reciprocal value.
 
         Args:
-            risk_allele (Column): Risk allele column
-            reference_allele (Column): Reference allele column
-            alternate_allele (Column): Alternate allele column
-            effect_size (Column): GWAS Catalog effect size column
-            confidence_interval (Column): GWAS Catalog confidence interval column
+            effect_size (Column): containing effect size,
+            flipping_needed (Column): Boolean flag indicating if effect needs to be flipped
 
         Returns:
             Column: A column with the odds ratio, or 1/odds_ratio if harmonization required.
+
+        Examples:
+        >>> d = [(0.5, False), (0.5, True), (0.0, False), (0.0, True)]
+        >>> (
+        ...    spark.createDataFrame(d, ['effect', 'flip'])
+        ...    .select("effect", "flip", GWASCatalogCuratedAssociationsParser._harmonise_odds_ratio(f.col("effect"), f.col("flip")).alias("odds_ratio"))
+        ...    .show()
+        ... )
+        +------+-----+----------+
+        |effect| flip|odds_ratio|
+        +------+-----+----------+
+        |   0.5|false|       0.5|
+        |   0.5| true|       2.0|
+        |   0.0|false|       0.0|
+        |   0.0| true|      null|
+        +------+-----+----------+
+        <BLANKLINE>
         """
         return (
-            f.when(
-                GWASCatalogCuratedAssociationsParser._are_alleles_palindromic(
-                    reference_allele, alternate_allele
-                ),
-                None,
-            )
+            # We are not flipping zero effect size:
+            f.when((effect_size.cast(DoubleType()) == 0) & flipping_needed, f.lit(None))
```

Comment:
Might be missing something here, but isn't this function dealing with ORs? OR = 0 shows a protective effect.

Reply:
OR=0 means no effect. Most likely we would not ever see association with 0 odds ratio. However, if that's happening, I would just keep it, but the effect cannot be flipped, because it would involve a division with zero.

Reply:
Of course, didn't occur to me you were addressing division by 0 issues thanks! I agree that I can't see how a GWAS could have such effect size, however maybe it makes more sense mathematically speaking to just assign a very low value like 0.001 instead of a null?